assets/src/js/completed_actions.js

```diff
@@ -0,0 +1,56 @@
+const {__} = wp.i18n;
+
+export const markActionsAsCompleted = () => {
+  // Make sure customization is enabled in the settings
+  const isCustomizationEnabled = Boolean(window.p4_vars.features.actions_user_personalization);
+  if (!isCustomizationEnabled) {
+    return;
+  }
+
+  // Make sure that we are in a Resistance Hub template
+  if (!document.body.classList.contains('p4_action-template-single-p4_action-resistance_hub')) {
+    return;
+  }
+
+  // Initialise completed actions in local storage if needed.
+  const completedActions = localStorage.completedActions;
```

Comment:
I think you should use `getItem` or `setItem` as an official window API implementation.

For instance

```js
const completedActions = "p4-completed-actions";
if(!localStorage.getItem(completedActions)) {
  localStorage.setItem(completedActions);
}

const ids = localStorage.getItem(completedActions).split(",");
```